src/platforms/generic_unix/CMakeLists.txt

```diff
@@ -18,58 +18,43 @@
 # SPDX-License-Identifier: Apache-2.0 OR LGPL-2.1-or-later
 #
 
-cmake_minimum_required (VERSION 2.6)
-project (libAtomVMPlatformGenericUnix)
+cmake_minimum_required (VERSION 3.13)
 
-include_directories(${CMAKE_CURRENT_BINARY_DIR} ${CMAKE_SOURCE_DIR}/src/libAtomVM)
+add_executable(AtomVM main.c)
+set_target_properties(AtomVM PROPERTIES RUNTIME_OUTPUT_DIRECTORY ../../)
+target_compile_features(AtomVM PUBLIC c_std_11)
+if(CMAKE_COMPILER_IS_GNUCC)
+    target_compile_options(AtomVM PUBLIC -Wall -pedantic -Wextra -ggdb)
+endif()
 
-set(HEADER_FILES
-    generic_unix_sys.h
-    mapped_file.h
-    platform_defaultatoms.h
-)
+if(${CMAKE_SYSTEM_NAME} STREQUAL "Linux")
+    include(CheckLibraryExists)
+    check_library_exists(rt clock_gettime "time.h" HAVE_CLOCK_GETTIME)
```

Comment:
Style question: I realize this is a small thing, but would it make more sense for checks like this to be done in a module in the top-level CMakeModules directory?  I don't really have an answer to that.  Just wondering if that is better "crake-style", if that is a thing.

Reply:
OTOH, this check really is for Unix. I tried to put such checks at the top CMakeLists.txt for a given platform.